### ISSUE TITLE
Deprecate the pyomo install-extras subcommand

### ISSUE THREAD

**Issue by whart222** [Pyomo/pyomo#435]
The conda pyomo.extras package supports this functionality more robustly.  We should not duplicate this logic in separate places.

**Comment by jsiirola** [comment]
I am not sure I agree.  If we are going to continue to support an "extras" (optional dependencies) package, I feel we should support both conda and a non-conda option.  Ideally the conda recipe would query the code behind the `pyomo install-extras` command to ensure that they install the same packages.

**Comment by whart222** [comment]
I'm not sure what you mean by "the conda recipe would query the code behind the pyomo install-extras command".  The conda-forge build script builds a conda package from a YAML metafile.  I don't see how this would ever be derived or be related to the pyomo logic.

**Comment by whart222** [comment]
Fundamentally, the pyomo install-extras logic is different from conda because we found that creating a Python package with simple dependencies for pyomo.extras was not robust.  At least a while back, pip would routinely fail for some of Pyomo's optional dependencies.  Hence, we tried out conda, which has proved more robust.  The logic executed by pyomo install-extras tries to manually install packages such that failures are handled in a graceful manner.  Maybe pip can does this itself now-adays.  But in any case these seem like separate processes to achieve the same result.

I'm still inclined to tell users that we support the easy installation of extras with just conda.  This doesn't limit *them* from installing these dependencies themselves.
